File: magellan-sample-migration/src/main/java/com/wealthfront/magellan/sample/migration/AppModule.java
package com.wealthfront.magellan.sample.migration;

import com.wealthfront.magellan.navigation.NavigationTraverser;
import com.wealthfront.magellan.sample.migration.api.DogApi;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import io.reactivex.schedulers.Schedulers;
import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Retrofit;
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;
import retrofit2.converter.jackson.JacksonConverterFactory;

@Module
final class AppModule {

  private static final String DOG_BASE_URL = "https://dog.ceo/api/";

  @Provides
  @Singleton
  Expedition provideExpedition() {
    return new Expedition();
  }

  @Provides
  @Singleton
  NavigationTraverser provideNavigationTraverser(Expedition root) {
    return new NavigationTraverser(root);
  }

  @Provides
  @Singleton
  DogApi provideDogApi(Retrofit retrofit) {
    return retrofit.create(DogApi.class);
  }

  @Provides
  @Singleton
  Retrofit provideRetrofit(OkHttpClient httpClient) {
    return new Retrofit.Builder()
        .baseUrl(DOG_BASE_URL)
        .addCallAdapterFactory(RxJava2CallAdapterFactory.createWithScheduler(Schedulers.io()))
        .addConverterFactory(JacksonConverterFactory.create())
        .client(httpClient)
        .build();
  }

  @Provides
  @Singleton
  OkHttpClient provideHttpClient() {
    HttpLoggingInterceptor interceptor = new HttpLoggingInterceptor();
    interceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
    return new OkHttpClient.Builder().addInterceptor(interceptor).build();
  }

}

File: magellan-sample-migration/src/main/java/com/wealthfront/magellan/sample/migration/AppModule.kt
package com.wealthfront.magellan.sample.migration

import com.wealthfront.magellan.navigation.NavigationTraverser
import com.wealthfront.magellan.sample.migration.api.RequestCountingInterceptor
import dagger.Module
import dagger.Provides
import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.converter.jackson.JacksonConverterFactory
import javax.inject.Singleton

private const val DOG_BASE_URL = "https://dog.ceo/api/"

@Module
object AppModule {

  @Provides
  @Singleton
  fun provideExpedition(): Expedition {
    return Expedition()
  }

  @Provides
  @Singleton
  fun provideNavigationTraverser(root: Expedition): NavigationTraverser {
    return NavigationTraverser(root)
  }

  @Provides
  @Singleton
  fun provideRetrofit(httpClient: OkHttpClient): Retrofit {
    return Retrofit.Builder()
      .baseUrl(DOG_BASE_URL)
      .addConverterFactory(JacksonConverterFactory.create())
      .client(httpClient)
      .build()
  }

  @Provides
  @Singleton
  fun provideHttpClient(): OkHttpClient {
    val interceptor = HttpLoggingInterceptor()
    interceptor.setLevel(HttpLoggingInterceptor.Level.BODY)
    return OkHttpClient.Builder()
      .addInterceptor(interceptor)
      .addInterceptor(RequestCountingInterceptor())
      .build()
  }
}
